app/Blockers.cs

```diff
@@ -0,0 +1,158 @@
+                }
+                
+                db.Blockers.Add(newBlock);
+                db.SaveChanges();
+                break;
+            case Verb.Edit:
+                // check if the option is invalid.
+                if(options.ContainsKey(Option.Invalid)) {
+                    throw new ArgumentException("Invalid option.");
+                }
+
+                // check if the options has blocker id
+                if((!options.ContainsKey(Option.BlockerId)) || (options.ContainsKey(Option.BlockerId) && options[Option.BlockerId].Count() == 0)) {
+                    throw new ArgumentException("BlockerId is needed to edit the blocker.");
+                }
+
+                Block existBlock = db.Blockers.Where(b => b.Id == Convert.ToInt32(options[Option.BlockerId][0])).FirstOrDefault();
+                if(existBlock == null) {
+                    throw new ArgumentException("Wrong BlockerId was entered.");
+                }
+
+                // if options have description, change the block's description to new value.
+                if(options.ContainsKey(Option.Description)) {
+                    foreach(string des in options[Option.Description]) {
+                        existBlock.Description = des;
+                    }
+                }
+
+                // if options have workitem list, change the block's workitems to new value.
+                if(options.ContainsKey(Option.WorkItemId)) {
+                    List<WorkItem> workItems = new List<WorkItem>();
+                    foreach(string workitemid in options[Option.WorkItemId]) {
+                        WorkItem workItem = db.WorkItems.Where(w => w.Id == Convert.ToInt32(workitemid)).FirstOrDefault();
+                        workItems.Add(workItem);
+                    }
+                    existBlock.WorkItems = workItems;
+                }
+
+                db.SaveChanges();
+                break;
+            case Verb.Delete:
+                // check if the option is invalid.
+                if(options.ContainsKey(Option.Invalid)) {
+                    throw new ArgumentException("Invalid option.");
+                }
+
+                // check if the options has blocker id
+                if((!options.ContainsKey(Option.BlockerId)) || (options.ContainsKey(Option.BlockerId) && options[Option.BlockerId].Count() == 0)) {
+                    throw new ArgumentException("BlockerId is needed to delete the blocker.");
+                }
+
+                Block blockToDelete = db.Blockers.Where(b => b.Id == Convert.ToInt32(options[Option.BlockerId][0])).FirstOrDefault();
+                if(blockToDelete == null) {
+                    throw new ArgumentException("Wrong BlockerId was entered.");
+                }
+
+                db.Blockers.Remove(blockToDelete);
+                db.SaveChanges();
+                break;
+            default:
+                throw new ArgumentException("Invalid verb.");
+        }
+    }
```

Comment:
I think we should clean up this function a little bit. See my earlier comment about using the order of arguments to determine what the argument is.

Also, I think we should split up the `nouns` into different helper functions. This could help us with testing.

I definitely see that I could've given more direction :( sorry for the confusion!